samza-core/src/main/java/org/apache/samza/task/TaskCallbackManager.java

```diff
@@ -76,7 +76,7 @@ List<TaskCallbackImpl> update(TaskCallbackImpl cb) {
   private final TaskCallbacks completedCallbacks = new TaskCallbacks();
   private final ScheduledExecutorService timer;
   private final TaskCallbackListener listener;
-  private final long timeout;
+  private long timeout;
```

Comment:
In general, its anti pattern to override instance variables post construction. We should probably find another way to enforce timeout for drain.

Reply:
Fixed this.